Include advice about dispose in TextEditingController api

rubensdemelo opened the pull request:
The same note in https://flutter.dev/docs/cookbook/forms/text-field-changes

## Related Issues

Fixes https://github.com/flutter/flutter/issues/29324

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [X] No, this is *not* a breaking change.

rubensdemelo committed:
include advice about dispose

The same note in https://flutter.dev/docs/cookbook/forms/text-field-changes

HansMuller reviewed (commented):
This is a good change! It would be helpful to provide the message about managing the lifetime of the TextEditingController to the other widgets that have a TextEditingController parameter: TextField, TextFormField, CupertinoTextField. Incorporating a link to https://flutter.dev/docs/cookbook/forms/text-field-changes#2-use-a-texteditingcontroller would also be helpful.

rubensdemelo committed:
Include remember to dispose texteditingcontroller

Almost the same note here https://flutter.dev/docs/cookbook/forms/text-field-changes#2-use-a-texteditingcontroller .

Piinks commented on packages/flutter/lib/src/material/text_field.dart:
```diff
@@ -62,6 +62,10 @@ typedef InputCounterWidgetBuilder = Widget Function(
+/// Remember to [dispose] the TextEditingController when it is no longer needed. 
+/// This will ensure we discard any resources used by the object. 
+/// 
+///
```
Nit: Remove extra line

rubensdemelo replied:
done

Piinks commented on packages/flutter/lib/src/material/text_field.dart:
```diff
@@ -62,6 +62,10 @@ typedef InputCounterWidgetBuilder = Widget Function(
 /// tapped an ink splash that paints on the material is triggered, see
 /// [ThemeData.splashFactory].
 ///
+/// Remember to [dispose] the TextEditingController when it is no longer needed. 
```
```suggestion
/// Remember to [dispose] of the [TextEditingController] when it is no longer needed. 
```

Here and below.

rubensdemelo replied:
done

Piinks reviewed (commented):
The analyzer seems to have detected some extra white space at the end of 1+ of these lines.

rubensdemelo committed:
include remember to dispose TextEditingController and link to see also

rubensdemelo committed:
remove extras spaces and duplicated comment

rubensdemelo committed:
include details about using Positioned.fill

rubensdemelo committed:
Revert "include details about using Positioned.fill"

This reverts commit c06196f8884d20c547d97edae71d26f7a61701be.

rubensdemelo committed:
rever comment about Positioned.fill

Piinks commented on packages/flutter/lib/src/cupertino/text_field.dart:
```diff
@@ -116,6 +116,9 @@ enum OverlayVisibilityMode {
 /// rounded rectangle border around the text field. If you set the [decoration]
 /// property to null, the decoration will be removed entirely.
+/// 
+/// Remember to [dispose] the [TextEditingController] when it is no longer needed. 
```
```suggestion
/// Remember to [dispose] of the [TextEditingController] when it is no longer needed.
```

rubensdemelo replied:
done

Piinks commented on packages/flutter/lib/src/cupertino/text_field.dart:
```diff
@@ -124,6 +127,7 @@ enum OverlayVisibilityMode {
 ///    Design UI conventions.
 ///  * [EditableText], which is the raw text editing control at the heart of a
 ///    [TextField].
+///  * <https://flutter.dev/docs/cookbook/forms/text-field-changes#2-use-a-texteditingcontroller>
```
Linking to a cookbook example is a good idea. I suggest making it a bit more readable:

```suggestion
///  * Learn how to use a [TextEditingController] in one of our [cookbook recipe]s.(https://flutter.dev/docs/cookbook/forms/text-field-changes#2-use-a-texteditingcontroller)
```

Here and below.

rubensdemelo replied:
done

Piinks commented on packages/flutter/lib/src/material/text_field.dart:
```diff
@@ -64,6 +64,9 @@ typedef InputCounterWidgetBuilder = Widget Function(
 /// To integrate the [TextField] into a [Form] with other [FormField] widgets,
 /// consider using [TextFormField].
+/// 
+/// Remember to [dispose] the [TextEditingController] when it is no longer needed. 
```
```suggestion
/// Remember to [dispose] of the [TextEditingController] when it is no longer needed.
```

rubensdemelo replied:
done

Piinks commented on packages/flutter/lib/src/material/text_form_field.dart:
```diff
@@ -28,6 +28,9 @@ import 'theme.dart';
 /// If a [controller] is not specified, [initialValue] can be used to give
 /// the automatically generated controller an initial value.
+/// 
+/// Remember to [dispose] the [TextEditingController] when it is no longer needed. 
```
```suggestion
/// Remember to [dispose] of the [TextEditingController] when it is no longer needed.
```

rubensdemelo replied:
done

Piinks commented on packages/flutter/lib/src/widgets/editable_text.dart:
```diff
@@ -65,7 +65,9 @@ const int _kObscureShowLatestCharCursorTicks = 3;
 /// controller's [value] should be set instead.
-///
+/// 
+/// Remember to [dispose] the [TextEditingController] when it is no longer needed. 
```
```suggestion
/// Remember to [dispose] of the [TextEditingController] when it is no longer needed.
```

rubensdemelo replied:
done

Piinks commented on packages/flutter/lib/src/material/text_field.dart:
```diff
@@ -91,6 +94,7 @@ typedef InputCounterWidgetBuilder = Widget Function(
 ///    [TextField]. The [EditableText] widget is rarely used directly unless
 ///    you are implementing an entirely different design language, such as
 ///    Cupertino.
+///  * <https://flutter.dev/docs/cookbook/forms/text-field-changes#2-use-a-texteditingcontroller>
```
Linking to a cookbook example is a good idea. I suggest making it a bit more readable:
```suggestion
///  * Learn how to use a [TextEditingController] in one of our [cookbook recipe]s.(https://flutter.dev/docs/cookbook/forms/text-field-changes#2-use-a-texteditingcontroller)
```

rubensdemelo replied:
done

Piinks commented on packages/flutter/lib/src/material/text_form_field.dart:
```diff
@@ -60,6 +63,7 @@ import 'theme.dart';
 ///    integration.
 ///  * [InputDecorator], which shows the labels and other visual elements that
 ///    surround the actual text editing widget.
+///  * <https://flutter.dev/docs/cookbook/forms/text-field-changes#2-use-a-texteditingcontroller>
```
Linking to a cookbook example is a good idea. I suggest making it a bit more readable:
```suggestion
///  * Learn how to use a [TextEditingController] in one of our [cookbook recipe]s.(https://flutter.dev/docs/cookbook/forms/text-field-changes#2-use-a-texteditingcontroller)
```

rubensdemelo replied:
done

Piinks commented on packages/flutter/lib/src/widgets/editable_text.dart:
```diff
@@ -112,6 +114,7 @@ const int _kObscureShowLatestCharCursorTicks = 3;
 ///    with a [TextEditingController].
 ///  * [EditableText], which is a raw region of editable text that can be
 ///    controlled with a [TextEditingController].
+///  * <https://flutter.dev/docs/cookbook/forms/text-field-changes#2-use-a-texteditingcontroller>
```
Linking to a cookbook example is a good idea. I suggest making it a bit more readable:
```suggestion
///  * Learn how to use a [TextEditingController] in one of our [cookbook recipe]s.(https://flutter.dev/docs/cookbook/forms/text-field-changes#2-use-a-texteditingcontroller)
```

rubensdemelo replied:
@Piinks Thank you for reviewing. Already make that changes.

Piinks reviewed (commented):
Hi @rubensdemelo,
Thank you for your continued work on this!
One of the things that is causing our analyzer to fail is extra white space at the end of some of the lines you have written, I've marked some of them along with a few other nits below.

rubensdemelo committed:
Include of and remove extra white space

Include of and remove extra white space in the end of line

rubensdemelo committed:
improve link description to cookbook

///  * Learn how to use a [TextEditingController] in one of our [cookbook recipe]s.(https://flutter.dev/docs/cookbook/forms/text-field-changes#2-use-a-texteditingcontroller)

rubensdemelo committed:
include of, remove extra white space and improve description link to cookbook.

include of, remove extra white space and improve description link to cookbook.

rubensdemelo committed:
include of, remove extra white space and improve link description to cookbook

include of, remove extra white space and improve link description to cookbook

rubensdemelo committed:
include of, remove extra white space and improve link description to cookbook

include of, remove extra white space and improve link description to cookbook

rubensdemelo commented:
@Piinks Thank you for reviewing. Already make that changes.

Piinks commented on packages/flutter/lib/src/widgets/editable_text.dart:
```diff
@@ -65,7 +65,9 @@ const int _kObscureShowLatestCharCursorTicks = 3;
 /// added to this controller. If both properties need to be changed then the
 /// controller's [value] should be set instead.
-///
+/// 
```
```suggestion
///
```
whitespace

Piinks reviewed (commented):
Hi @rubensdemelo,
Thank you for sticking with this. The local `flutter analyze --flutter-repo` tool doesn't catch these pesky, invisible little whitespaces. It looks like there a few more hiding here that are causing the analyzer to fail.

rubensdemelo committed:
Merge branch 'master' of github.com:rubensdemelo/flutter

rubensdemelo committed:
remove extra white spaces

rubensdemelo commented:
Hi @Piinks ! Thank you for helping me. I already remove extras white spaces, but it seems `analyze` still fails: 
```
packages/flutter/lib/src/material/text_field.dart packages/flutter/lib/src/material/text_form_field.dart packages/flutter/lib/src/widgets/basic.dart packages/flutter/lib/src/widgets/editable_text.dart in .: 
Whitespace detected at the end of source code lines.
Please remove:
```

rubensdemelo committed:
remove extra white space

Piinks reviewed (approved):
LGTM! Thanks again for wading through the nits with me! I'll land this once the build is green! 
Cheers! 🎉
